internal/tools/aws/database_multitenant.go

```diff
@@ -0,0 +1,637 @@
+	}
+
+	installationSecreteName := RDSMultitenantSecretName(d.installationID)
+
+	result, err := d.client.Service().secretsManager.GetSecretValue(&secretsmanager.GetSecretValueInput{
+		SecretId: &installationSecreteName,
+	})
+	if err != nil {
+		return nil, nil, errors.Wrap(err, "unable to generate databa spec and secret")
+	}
+
+	installationSecret, err := unmarshalSecretPayload(*result.SecretString)
+	if err != nil {
+		return nil, nil, errors.Wrap(err, "unable to generate databa spec and secret")
+	}
+
+	installationDBName := MattermostRDSDatabaseName(d.installationID)
+	installationDBConnString := MattermostMySQLConnString(installationDBName, *rdsCluster.Endpoint, installationSecret.MasterUsername, installationSecret.MasterPassword)
+
+	databaseSecret := &corev1.Secret{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: installationSecreteName,
+		},
+		StringData: map[string]string{
+			"DB_CONNECTION_STRING": installationDBConnString,
+		},
+	}
+
+	databaseSpec := &mmv1alpha1.Database{
+		Secret: installationSecreteName,
+	}
+
+	logger.Infof("Finish to set up spec and secret for database %s", installationDBName)
+	logger.Debugf("RDS DB cluster %s configured to use connection string %s", *rdsCluster.DBClusterIdentifier, installationDBConnString)
+
+	return databaseSpec, databaseSecret, nil
+}
+
+// Provision claims a multitenant RDS cluster and creates a database schema for the installation.
+func (d *RDSMultitenantDatabase) Provision(store model.InstallationDatabaseStoreInterface, logger log.FieldLogger) error {
+	databaseName := MattermostRDSDatabaseName(d.installationID)
+	logger = logger.WithField("multitenant-rds-database", databaseName)
+
+	vpc, err := d.getClusterInstallationVPC(store)
+	if err != nil {
+		return errors.Wrapf(err, "unable to find the VPC for installation ID %s", d.installationID)
+	}
+
+	lockedCluster, err := d.lockRDSCluster(*vpc.VpcId, store, logger)
+	if err != nil {
+		return errors.Wrapf(err, "unable to lock a RDS cluster for installation ID %s", d.installationID)
+	}
+	defer lockedCluster.unlock(logger)
+
+	masterSecretValue, err := d.client.Service().secretsManager.GetSecretValue(&secretsmanager.GetSecretValueInput{
+		SecretId: lockedCluster.cluster.DBClusterIdentifier,
+	})
+	if err != nil && !IsErrorCode(err, secretsmanager.ErrCodeResourceNotFoundException) {
+		return errors.Wrapf(err, "unable to find the master secret for RDS cluster ID %s", *lockedCluster.cluster.DBClusterIdentifier)
+	}
+
+	close, err := d.connectRDSCluster(rdsMySQLSchemaInformationDatabase, *lockedCluster.cluster.Endpoint, DefaultMattermostDatabaseUsername, *masterSecretValue.SecretString)
+	defer close(logger)
+	if err != nil {
+		return errors.Wrapf(err, "unable to connect to RDS cluster ID %s", *lockedCluster.cluster.DBClusterIdentifier)
+	}
+
+	ctx, cancel := context.WithDeadline(context.Background(), time.Now().Add(DefaultMySQLDefaultContextTimeout*time.Second))
```

Comment:
2 things:
1) `DefaultMySQLDefaultContextTime` has `Default` in it twice and I'm not sure that was intentional.
2) Should `DefaultMySQLDefaultContextTime` be a `time` type instead of an `int` so that the unit of time measured in the actual default is unambiguous and the consumer of that object doesn't need to know that it's in seconds?

Reply:
You can't create constants of complex types in Go, unfortunately. The double Default is not intentional.

Reply:
Does it need to be an actual `const`? I'd rather have a `var` of the correct type personally, but I guess that's personal preference. If you don't like that, perhaps include the unit as part of the name, e.g. `DefaultMySQLContextTimeSeconds`?

Reply:
The constant unit useful if we want to fine tune it in the future. We don't need to change every single call site.

Reply:
I can change it to `DefaultMySQLContextTimeSeconds`.